src/AWS/Lambda/KinesisDataStreamsEvent.hs

```diff
@@ -1,69 +1,56 @@
-{-# LANGUAGE DeriveGeneric     #-}
-{-# LANGUAGE OverloadedStrings #-}
-{-# LANGUAGE RecordWildCards   #-}
-
-module AWS.Lambda.KinesisDataStreamsEvent
-  (
-    KinesisDataStreamsEvent(..),
-    Record(..),
-    Kinesis(..)
-  ) where
+{-# language DeriveGeneric   #-}
+{-# language RecordWildCards #-}
+{-# language TemplateHaskell #-}
+module AWS.Lambda.KinesisDataStreamsEvent where
 
+import Control.Lens
 import Data.Aeson
 import Data.HashMap.Strict
 import Data.Text (Text)
 import GHC.Generics
 
-data Kinesis =
-  Kinesis {
-    partitionKey         :: Text,
-    payload              :: Text,
-    kinesisSchemaVersion :: Text,
-    sequenceNumber       :: Text
-  } deriving (Show)
+data Kinesis = Kinesis
+  { _partitionKey         :: Text
+  , _payload              :: Text
+  , _kinesisSchemaVersion :: Text
+  , _sequenceNumber       :: Text
+  } deriving (Eq, Generic, Show)
+
+instance ToJSON Kinesis where
+  toJSON = genericToJSON defaultOptions { fieldLabelModifier = drop 1 }
 
 instance FromJSON Kinesis where
-  parseJSON (Object v) = Kinesis
-    <$> v .: "partitionKey"
-    <*> v .: "data"
-    <*> v .: "kinesisSchemaVersion"
-    <*> v .: "sequenceNumber"
+  parseJSON = genericParseJSON defaultOptions { fieldLabelModifier = drop 1 }
 
-instance ToJSON Kinesis where
-  toJSON Kinesis{..} =
-    object [
-        "partitionKey" .= partitionKey
-      , "data" .= payload
```

Comment:
Notice that what we called `payload` in `Kinesis` is serialized/deserialized as "data". We couldn't use the `data` keyword as it's a reserved keyword in the language